profiles/seccomp/seccomp.go

```diff
@@ -6,28 +6,64 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/versions/v1p24"
 	"github.com/opencontainers/specs/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
 )
 
 //go:generate go run -tags 'seccomp' generate.go
 
 // GetDefaultProfile returns the default seccomp profile.
 func GetDefaultProfile(rs *specs.Spec) (*specs.Seccomp, error) {
-	return setupSeccomp(DefaultProfile(rs))
+	return setupSeccomp(DefaultProfile(), rs)
 }
 
 // LoadProfile takes a file path and decodes the seccomp profile.
-func LoadProfile(body string) (*specs.Seccomp, error) {
-	var config types.Seccomp
+func LoadProfile(body string, rs *specs.Spec) (*specs.Seccomp, error) {
+	var (
+		config types.Seccomp
+		err    error
+	)
 	if err := json.Unmarshal([]byte(body), &config); err != nil {
 		return nil, fmt.Errorf("Decoding seccomp profile failed: %v", err)
 	}
+	// try with the old profile
+	// TODO(runcom): this is not going to work properly:
+	// if I use Version == 0 in a new format seccomp profile, what's going to
+	// happen?
+	if config.Version == 0 {
```

Comment:
> er, if we allow name or names, then all version 0 json is valid as version 1, so we can always parse as version 1? Is this correct?

yes, unless you come with version 0 and `names` only - case 1) and 2)
and yes again but I'd prefer to leave the `decodeLegacySeccomp`, so we can catch 1) and 2) because we'll have that "empty syscall name" error (related to https://github.com/docker/docker/pull/24537#issuecomment-232075616 since ppl miss "name" there)